tfx/experimental/distributed_inference/graphdef_experiments/subgraph_partitioning/beam_pipeline.py

```diff
@@ -0,0 +1,334 @@
+    yield element
+
+
+def _import_tensor_name(node_name: Text) -> Text:
+  return 'import/%s:0' % node_name
+
+
+@beam.ptransform_fn
+@beam.typehints.with_input_types(Dict[Text, Dict[Text, tf.Tensor]])
+@beam.typehints.with_output_types(Dict[Text, Dict[Text, tf.Tensor]])
+def _LoadRemoteGraphInputs(  # pylint: disable=invalid-name
+    pcoll: beam.pvalue.PCollection,
+    parent_graph_name_extended: Text,
+    remote_graph_name_extended: Text,
+    parent_graph_to_remote_graph_input_name_mapping:
+        Mapping[Text, Mapping[Text, Mapping[Text, Text]]],
+    graph_names: Set[Text]) -> beam.pvalue.PCollection:
+  """Loads remote graph inputs.
+
+  Before executing a remote graph, we need to prepare for its inputs. We first
+  get the mapping from remote graph placeholder names to parent graph input
+  names. Then, we copy the inputs from the parent graph to the remote graph
+  in element.
+
+  Args:
+    pcoll: A number of elements, where remote graph inputs aren't seted up.
+    parent_graph_name_extended: The extended name of the parent graph.
+    remote_graph_name_extended: The extended name of the remote graph.
+    parent_graph_to_remote_graph_input_name_mapping:
+      A mapping from graph names to remote graph name extendeds to remote
+      graph placeholder names to parent graph input names.
+      {graph name: {remote graph extended: {placeholder name: input name}}}.
+    graph_names: A set of all the graph names.
+
+  Returns:
+    A number of elements, where remote graph's inputs are seted up.
+  """
+  parent_graph_name = _get_graph_name(parent_graph_name_extended, graph_names)
+  name_mapping = (parent_graph_to_remote_graph_input_name_mapping
+                  [parent_graph_name][remote_graph_name_extended])
+
+  mapping = name_mapping.items()
+  # Calling _copy_tensor multiple times may introduce a burden, since
+  # _copy_tensor invokes a deepcopy on element.
+  for remote_graph_placeholder_name, parent_graph_input_name in mapping:
+
+    step_name = "Prepare input %s in %s" % (
+        remote_graph_placeholder_name, remote_graph_name_extended)
+    pcoll = pcoll | step_name >> beam.Map(
+        _copy_tensor,
+        parent_graph_name_extended,
+        _import_tensor_name(parent_graph_input_name),
+        remote_graph_name_extended,
+        _import_tensor_name(remote_graph_placeholder_name))
+
+  return pcoll
+
+
+def _copy_tensor(
+    element: Dict[Text, Dict[Text, tf.Tensor]],
+    old_graph: Text, old_tensor_name: Text,
+    new_graph: Text, new_tensor_name: Text
+) -> Dict[Text, Dict[Text, tf.Tensor]]:
+  element = copy.deepcopy(element)
+  if new_graph not in element:
+    element[new_graph] = {}
+  element[new_graph][new_tensor_name] = element[old_graph][old_tensor_name]
+  return element
+
+
+@beam.ptransform_fn
+@beam.typehints.with_input_types(Dict[Text, Dict[Text, tf.Tensor]])
+@beam.typehints.with_output_types(Dict[Text, Dict[Text, tf.Tensor]])
+def _ExtractRemoteGraphOutput(  # pylint: disable=invalid-name
```

Comment:
Same comments here as for `_LoadRemoteGraphInputs`.

Reply:
Addressed.